application/src/main/java/org/thingsboard/server/service/security/auth/jwt/settings/JwtSettingsServiceDefault.java

```diff
@@ -0,0 +1,169 @@
+    private Integer tokenExpirationTime;
+    @Value("${security.jwt.refreshTokenExpTime:604800}")
+    private Integer refreshTokenExpTime;
+    @Value("${security.jwt.tokenIssuer:thingsboard.io}")
+    private String tokenIssuer;
+    @Value("${security.jwt.tokenSigningKey:thingsboardDefaultSigningKey}")
+    private String tokenSigningKey;
+
+    @PostConstruct
+    public void init() {
+        jwtSettings.setTokenExpirationTime(this.tokenExpirationTime);
+        jwtSettings.setRefreshTokenExpTime(this.refreshTokenExpTime);
+        jwtSettings.setTokenIssuer(this.tokenIssuer);
+        jwtSettings.setTokenSigningKey(this.tokenSigningKey);
+        if (!isFirstInstall()) {
+            reloadJwtSettings();
+        }
+    }
+
+    private boolean isInstall() {
+        return environment.acceptsProfiles(Profiles.of("install"));
+    }
+
+    private boolean isFirstInstall() {
+        return isInstall() && !isUpgrade;
+    }
+
+    @Override
+    public void reloadJwtSettings() {
+        AdminSettings adminJwtSettings = findJwtAdminSettings();
+        if (adminJwtSettings != null) {
+            log.info("Reloading the JWT admin settings from database");
+            JwtSettings jwtLoaded = mapAdminToJwtSettings(adminJwtSettings);
+            jwtSettings.setTokenExpirationTime(jwtLoaded.getTokenExpirationTime());
+            jwtSettings.setRefreshTokenExpTime(jwtLoaded.getRefreshTokenExpTime());
+            jwtSettings.setTokenIssuer(jwtLoaded.getTokenIssuer());
+            jwtSettings.setTokenSigningKey(jwtLoaded.getTokenSigningKey());
+        }
+
+        if (hasDefaultTokenSigningKey()) {
+            log.warn("WARNING: The platform is configured to use default JWT Signing Key. " +
+                    "This is a security issue that needs to be resolved. Please change the JWT Signing Key using the Web UI. " +
+                    "Navigate to \"System settings -> Security settings\" while logged in as a System Administrator.");
+        }
+    }
+
+    JwtSettings mapAdminToJwtSettings(AdminSettings adminSettings) {
+        Objects.requireNonNull(adminSettings, "adminSettings for JWT is null");
+        return JacksonUtil.treeToValue(adminSettings.getJsonValue(), JwtSettings.class);
+    }
+
+    AdminSettings mapJwtToAdminSettings(JwtSettings jwtSettings) {
+        Objects.requireNonNull(jwtSettings, "jwtSettings is null");
+        AdminSettings adminJwtSettings = new AdminSettings();
+        adminJwtSettings.setTenantId(TenantId.SYS_TENANT_ID);
+        adminJwtSettings.setKey(ADMIN_SETTINGS_JWT_KEY);
+        adminJwtSettings.setJsonValue(JacksonUtil.valueToTree(jwtSettings));
+        return adminJwtSettings;
+    }
+
+    boolean hasDefaultTokenSigningKey() {
+        return TOKEN_SIGNING_KEY_DEFAULT.equals(jwtSettings.getTokenSigningKey());
+    }
+
+    /**
+     * Create JWT admin settings is intended to be called from Install or Upgrade scripts
+     * */
+    @Override
+    public void createJwtAdminSettings() {
```

Comment:
This method is definitely misleading. Please create two methods:
```
saveRandomSettings();
saveLegacyYmlSettings();
```

In such a case the logic will be much more clear and have fewer chances of the bug.

Reply:
> This method is definitely misleading. Please create two methods:
> 
> ```
> saveRandomSettings();
> saveLegacyYmlSettings();
> ```
> 
> In such a case the logic will be much more clear and have fewer chances of the bug.

done as follow:

```
    void createRandomJwtSettings();
    void saveLegacyYmlSettings();
```